core/base/src/main/java/alluxio/underfs/UfsStatus.java

```diff
@@ -152,9 +159,17 @@ public String getOwner() {
     return mOwner;
   }
 
+  /**
+   * Returns the extended attributes from the Ufs, if any.
+   * @return a map of the extended attributes. If none, the map will be empty
+   */
+  public Map<String, String> getXAttr() {
```

Comment:
I am for using `byte[]` instead of string. I did some digging through the man pages (http://man7.org/linux/man-pages/man7/xattr.7.html and http://man7.org/linux/man-pages/man2/getxattr.2.html) and I didn't find anything specifying it had to be a string. I will convert to using `byte[]`

Reply:
Actually, I think we should want to be careful using `byte[]`. Proto's default representation of `bytes` in Java is as a `ByteString`. Unless we want to actively convert proto values, this will force inodes to depend on this proto-specific class, unless we perform conversion to `byte[]` actively whenever converting to/from proto. That could result in an increase in memory usage copying underlying byte arrays back and forth.

Reply:
We can store `byte[]` in inode and convert to/from `ByteString` when needed. 
The conversion between `ByteString` and `byte[]` won't be more expensive than the conversion between `String` and `byte[]` since creating a string involves copying byte arrays as well.

Reply:
Do we still want to update this class? If we do, should this now be `Map<String, byte[]>`?